src/sbl_filing_api/services/submission_processor.py

```diff
@@ -95,3 +95,29 @@ async def validate_and_update_submission(period_code: str, lei: str, submission:
         submission.state = SubmissionState.SUBMISSION_UPLOAD_MALFORMED
         await update_submission(submission)
         raise HTTPException(status_code=HTTPStatus.UNPROCESSABLE_ENTITY, detail=re)
+
+
+def build_validation_results(result):
+    val_json = json.loads(df_to_json(result[1]))
+    val_res = {}
+
+    if result[2] == ValidationPhase.SYNTACTICAL.value:
+        val_res["syntax_errors"] = {"count": 0, "details": []}
+        val_res["syntax_errors"]["details"] = val_json
+        val_res["syntax_errors"]["count"] = len(val_res["syntax_errors"]["details"])
+    else:
+        val_res = {
+            "syntax_errors": {"count": 0, "details": []},
+            "logic_errors": {"count": 0, "details": []},
+            "logic_warnings": {"count": 0, "details": []},
+        }
+        for v in val_json:
+            if v["validation"]["severity"] == Severity.WARNING.value:
+                val_res["logic_warnings"]["details"].append(v)
+            elif v["validation"]["severity"] == Severity.ERROR.value:
+                val_res["logic_errors"]["details"].append(v)
+        val_res["logic_warnings"]["count"], val_res["logic_errors"]["count"] = len(
+            val_res["logic_warnings"]["details"]
+        ), len(val_res["logic_errors"]["details"])
```

Comment:
using tuple destructuring here doesn't add much value here... you save 1 keystroke of not having to type out an additional `=`; but the formatting makes it a little more complicated to read.